magellan-library/src/test/java/com/wealthfront/magellan/navigation/LinearNavigatorTest.kt

```diff
@@ -121,17 +153,17 @@ class LinearNavigatorTest {
 
   @Test
   fun destroy() {
+    activityController.restart()
+    linearNavigator.resume(context)
     linearNavigator.navigate(FORWARD) {
       it.push(NavigationEvent(step1, GO))
       it.push(NavigationEvent(step2, GO))
       it.push(NavigationEvent(journey1, SHOW))
     }
 
-    linearNavigator.resume(getApplicationContext())
     linearNavigator.goBack()
     linearNavigator.goBack()
-    linearNavigator.pause(getApplicationContext())
-    linearNavigator.destroy(getApplicationContext())
+    linearNavigator.destroy(context)
```

Comment:
Does the navigator handle state transitions from e.g. `resumed` to `destroyed`? Does it transition its children through the proper lifecycle events?

Reply:
Since the linearNavigator is a lifecycleawarecomponent, that should be taken care of by the lifecycle registry.